Jenkinsfile.deploy

```
@@ -72,6 +85,38 @@ pipeline {
                 }
             }
         }
+        stage('Prepare for PR') {
+            when {
+                expression {
+                    return params.VERSION != '' && params.RELEASE
```

In the `Initialization` stage, check that `VERSION` is set if `RELEASE` is enabled and error if that is not the case.
Then you just need to have `params.RELEASE` here and you will be sure that VERSION is set :)

---

But like we can do this way also why to prior check it ?

---

because we should not accept the `RELEASE` true if no `VERSION` is set IMO

---

So in this also its && so in case of && , x&&y so if x becomes false then it will not even check for version thing and move forward in the stage.

---

We also want a fail fast in `Initialization` stage and then we don't need to check for both anymore ...

and no need to rewrite it many times as it is also the case here: https://github.com/kiegroup/kogito-images/pull/183/files#diff-4c6c57cd8094d1a0078f1cfa0e02fcf9R220

---

Okay cool I am getting it so just shifting the release thing in Initialization stage. :)